ansible_events/drools/durable_rules_engine.py

```diff
@@ -0,0 +1,230 @@
+        )
+        if r.status_code != 200:
+            raise Exception(
+                f"Invalid status code: {r.status_code} - {r.reason}\n"
+                + json.loads(r.content)["details"]
+            )
+
+        id = r.text
+        return id
+
+    def assert_event(self, session_id, serialized_fact):
+        _logger.warning("assert_event not yet implemented: using assert_fact")
+        return self.assert_fact(session_id, serialized_fact)
+
+    def assert_fact(self, session_id, serialized_fact):
+        d = json.loads(serialized_fact)
+        if "j" in serialized_fact:
+            d["j"] = 1
+
+        serialized_fact = json.dumps(d)
+
+        r = requests.post(
+            f"{self.__host}/rules-durable-executors/{session_id}/process",
+            json=json.loads(serialized_fact),
+        )
+
+        if r.status_code != 200:
+            raise Exception(
+                f"Invalid status code: {r.status_code} - {r.reason}\n"
+                + json.loads(r.content)["details"]
+            )
+
+        self.__last_resp = r.json()
+        self.__last_resp.reverse()
+
+        return (0, session_id)
+
+    def start_action_for_state(self, handle):  # real signature unknown
+        try:
+            resp = self.__last_resp.pop()
+        except Exception:
+            return None
+
+        return ('{ "sid":"0", "id":"sid-0", "$s":1}', json.dumps(resp), handle)
+
+    def complete_and_start_action(self, handle):  # real signature unknown
+        _logger.info("complete_and_start_action: %d", handle)
+        try:
+            resp = self.__last_resp.pop()
+        except Exception:
+            return None
+
+        return json.dumps(resp)
+
+    def retract_fact(self, session_id, serialized_fact):
+        r = requests.post(
+            f"{self.__host}/rules-durable-executors/{session_id}/retract-fact",
+            json=json.loads(serialized_fact),
+        )
+        if r.status_code != 200:
+            raise Exception(
+                f"Invalid status code: {r.status_code} - {r.reason}\n"
+                + json.loads(r.content)["details"]
+            )
+
+        return (0, session_id)
+
+    def get_facts(self, session_id, _sid):
+        r = requests.post(
+            f"{self.__host}/rules-durable-executors/{session_id}/get-all-facts"
+        )
+        if r.status_code != 200:
+            raise Exception(
+                f"Invalid status code: {r.status_code} - {r.reason}\n"
+                + json.loads(r.content)["details"]
+            )
+
+        return r.content
+
+
+class error(Exception):
+    # no doc
+    def __init__(self, *args, **kwargs):  # real signature unknown
+        pass
+
+    __weakref__ = property(
+        lambda self: object(), lambda self, v: None, lambda self: None
+    )  # default
+    """list of weak references to the object (if defined)"""
+
+
+# exported methods
+
+__instance = DurableRulesEngine("http://localhost:8080")
```

Comment:
Why double underscore prefix is used here?

Reply:
no special reason. Would a single underscore be better for a private global?

Reply:
using single underscore, added read from env var, so that is configurable